app/javascript/controllers/tab_error_controller.js

```diff
@@ -9,8 +9,17 @@ export default class extends Controller {
     // If it does, add the is-invalid class to the tab and modify the tab's
     // accessible label to include the error message.
     this.tabTargets.forEach((tabEl, index) => {
-      const paneEl = this.paneTargets[index]
-      if (paneEl.querySelector('.is-invalid')) {
+      const paneElement = this.paneTargets[index]
+      const invalidElements = paneElement.querySelectorAll('.is-invalid')
+      if (invalidElements.length > 0) {
+        invalidElements.forEach(invalidElement => {
+          invalidElement.addEventListener('change', event => {
+            const invalidFeedbackElement = paneElement.querySelector(`#${event.currentTarget.id}_error`)
+            event.currentTarget.classList.remove('is-invalid')
+            if (invalidFeedbackElement) invalidFeedbackElement.remove()
+            if (!paneElement.querySelector('.is-invalid')) tabEl.classList.remove('is-invalid')
+          })
+        })
```

Comment:
For each invalid element in the current pane, attach a change event listener whose responsibility is to:

1. Mark the field itself as no longer invalid
2. Destroy the associated invalid feedback `div`, if present
3. Mark the tab as no longer invalid if there are no remaining invalid elements in the pane